bumble/gatt_client.py

```diff
@@ -558,23 +555,32 @@ async def subscribe(self, characteristic, subscriber=None):
             logger.warning('subscribing to characteristic with no CCCD descriptor')
             return
 
-        # Set the subscription bits and select the subscriber set
-        bits = ClientCharacteristicConfigurationBits.DEFAULT
-        subscriber_sets = []
-        if characteristic.properties & Characteristic.NOTIFY:
-            bits |= ClientCharacteristicConfigurationBits.NOTIFICATION
-            subscriber_sets.append(self.notification_subscribers.setdefault(characteristic.handle, set()))
-        if characteristic.properties & Characteristic.INDICATE:
-            bits |= ClientCharacteristicConfigurationBits.INDICATION
-            subscriber_sets.append(self.indication_subscribers.setdefault(characteristic.handle, set()))
+        def characteristic_is_notify(characteristic, prefer_notify):
+            if (
+                characteristic.properties & Characteristic.NOTIFY
+                and characteristic.properties & Characteristic.INDICATE
+            ):
+                return prefer_notify
+            elif characteristic.properties & Characteristic.NOTIFY:
+                return True
+            elif characteristic.properties & Characteristic.INDICATE:
+                return False
+            else:
+                raise InvalidStateError('characteristic is not notify or indicate')
+
+        bits, subscribers = (
+            (ClientCharacteristicConfigurationBits.NOTIFICATION, self.notification_subscribers)
+            if characteristic_is_notify(characteristic, prefer_notify)
```

Comment:
i found this block (571-575) a bit hard to read. it's a tuple + if expression.
consider converting it to a if statement

Reply:
This may be a matter of opinion. I prefer the more succinct if-expression to the following if-statement.

```
bits = None
subscribers = None
if characteristic_is_notify(characteristic, prefer_notify):
    bits = ClientCharacteristicConfigurationBits.Notification
    subscribers = self.notification_subscribers
else:
    bits = ClientCharacteristicConfigurationBits.INDICATION
    subscribers = self.indication_subscribers
````

Since they both do exactly the same thing, I'll leave it as-is.

Reply:
You could also put the `bits` and `subscribers` assignments in the if/else statement used for `characteristic_is_notify` (and thus get rid of the local function). 
Also, since there's always at least one branch taken, you don't need to initialize `bits` and `subscribers` to `None`, they'll always be set  in one of the branches.